cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomalyFinder.java

```diff
@@ -0,0 +1,107 @@
+        if (partition.replicas().length != _targetReplicationFactor) {
+          topicsWithBadReplicationFactor.add(topic);
+          break;
+        }
+      }
+    }
+    if (!topicsWithBadReplicationFactor.isEmpty()) {
+      return Collections.singleton(createTopicReplicationFactorAnomaly(topicsWithBadReplicationFactor,
+                                                                       _targetReplicationFactor));
+    }
+    return Collections.emptySet();
+  }
+
+  private TopicAnomaly createTopicReplicationFactorAnomaly(Set<String> topicsWithBadReplicationFactor,
+                                                          int targetReplicationFactor) {
+    Map<String, Object> configs = new HashMap<>(4);
+    configs.put(KAFKA_CRUISE_CONTROL_OBJECT_CONFIG, _kafkaCruiseControl);
+    configs.put(TOPICS_WITH_BAD_REPLICATION_FACTOR_CONFIG, topicsWithBadReplicationFactor);
+    configs.put(SELF_HEALING_TARGET_TOPIC_REPLICATION_FACTOR_CONFIG, targetReplicationFactor);
+    configs.put(ANOMALY_DETECTION_TIME_MS_OBJECT_CONFIG, _kafkaCruiseControl.timeMs());
+    return getConfiguredInstance(_topicReplicationTopicAnomalyClass, TopicAnomaly.class, configs);
+  }
+
+  @Override
+  public void configure(Map<String, ?> configs) {
+    _kafkaCruiseControl = (KafkaCruiseControl) configs.get(KAFKA_CRUISE_CONTROL_OBJECT_CONFIG);
+    if (_kafkaCruiseControl == null) {
+      throw new IllegalArgumentException("Topic replication factor anomaly finder is missing " + KAFKA_CRUISE_CONTROL_OBJECT_CONFIG);
+    }
+    Short targetReplicationFactor = (Short) configs.get(SELF_HEALING_TARGET_TOPIC_REPLICATION_FACTOR_CONFIG);
+    if (targetReplicationFactor == null) {
+      throw new IllegalArgumentException("Topic replication factor anomaly finder is missing " + SELF_HEALING_TARGET_TOPIC_REPLICATION_FACTOR_CONFIG);
+    }
+    _targetReplicationFactor = targetReplicationFactor;
+    String topicExcludedFromCheck = (String) configs.get(TOPIC_EXCLUDED_FROM_REPLICATION_FACTOR_CHECK);
+    _topicExcludedFromCheck = Pattern.compile(topicExcludedFromCheck == null ? DEFAULT_TOPIC_EXCLUDED_FROM_REPLICATION_FACTOR_CHECK
+                                                                             : topicExcludedFromCheck);
+    String topicReplicationTopicAnomalyClass = (String) configs.get(TOPIC_REPLICATION_FACTOR_ANOMALY_CLASS_CONFIG);
+    if (topicReplicationTopicAnomalyClass == null) {
+      _topicReplicationTopicAnomalyClass = DEFAULT_TOPIC_REPLICATION_FACTOR_ANOMALY_CLASS;
+    } else {
+      _topicReplicationTopicAnomalyClass = (Class<?>) ConfigDef.parseType(TOPIC_REPLICATION_FACTOR_ANOMALY_CLASS_CONFIG,
+                                                                          topicReplicationTopicAnomalyClass,
+                                                                          CLASS);
+      if (_topicReplicationTopicAnomalyClass == null || !TopicAnomaly.class.isAssignableFrom(_topicReplicationTopicAnomalyClass)) {
```

Comment:
(Not introduced in this patch) We are missing similar sanity checks for pluggable `SlowBrokerFinder` for `AnomalyDetectorConfig.METRIC_ANOMALY_CLASS_CONFIG` .

Reply:
Discussed this offline, I would like to fix this in a follow-up patch to fix a bigger metric anomaly issue.